### PR TITLE
jamba liger fused linear+xentropy

### PR DESCRIPTION
**Pull request by winglian**
## Testing Done

- Hardware Type: <BLANK>
- [ ] run `make test` to ensure correctness
- [ ] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by ByronHsu** [comment]
awesome! please make sure you add both conv (w logits and w/o logits) and unit tests. we are very focused on testing

**Comment by ByronHsu** [comment]
https://github.com/linkedin/Liger-Kernel/issues/63

**Comment by yubofredwang** [comment]
I added the following additional monkey patch for Jamba.
```python
    from transformers.models.jamba import modeling_jamba
    if rms_norm:
        # https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/gemma/modeling_gemma.py#L109
        modeling_jamba.JambaRMSNorm = LigerRMSNorm
    if cross_entropy:
        modeling_jamba.CrossEntropyLoss = LigerCrossEntropyLoss
    if swiglu:
        modeling_jamba.JambaMLP = LigerSwiGLUMLP
```
However, convergence test seems to be failing for some values in the tensor:

```
E           Mismatch at index (0, 5): tensor1[(0, 5)] = 1.1513792276382446, tensor2[(0, 5)] = 1.1512681245803833
E           Mismatch at index (0, 27): tensor1[(0, 27)] = 0.6227690577507019, tensor2[(0, 27)] = 0.6227344870567322
E           Mismatch at index (0, 28): tensor1[(0, 28)] = 0.7790964841842651, tensor2[(0, 28)] = 0.7790292501449585
E           Mismatch at index (0, 29): tensor1[(0, 29)] = 0.524261474609375, tensor2[(0, 29)] = 0.5243569612503052
E           Mismatch at index (0, 30): tensor1[(0, 30)] = 0.8967938423156738, tensor2[(0, 30)] = 0.8968125581741333
```

I tracked this down to LigerRMSNorm but needs more time to investigate why there is a difference

File: src/liger_kernel/transformers/monkey_patch.py [hunk 1]
```diff
@@ -169,3 +169,41 @@ def apply_liger_kernel_to_qwen2(
...
+    assert not (
+            cross_entropy and fused_linear_cross_entropy
+    ), "cross_entropy and fused_linear_cross_entropy cannot both be True."
+
+    from transformers.models.jamba import modeling_jamba
+
+    if rope:
+        modeling_jamba.apply_rotary_pos_emb = liger_rotary_pos_emb
+    if rms_norm:
+        # https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/gemma/modeling_gemma.py#L109
```
Comment (ByronHsu) [inline src/liger_kernel/transformers/monkey_patch.py:204]: nit: the comment is wrong

File: src/liger_kernel/transformers/monkey_patch.py [hunk 2]
```diff
@@ -169,3 +169,41 @@ def apply_liger_kernel_to_qwen2(
...
+        modeling_jamba.JambaRMSNorm = LigerRMSNorm
+    if cross_entropy:
+        modeling_jamba.CrossEntropyLoss = LigerCrossEntropyLoss
+    if swiglu:
+        modeling_jamba.JambaMLP = LigerSwiGLUMLP
```
Comment (ByronHsu) [inline src/liger_kernel/transformers/monkey_patch.py:209]: where is lce_forward?

**Comment by yubofredwang** [comment]
HI @winglian created a PR towards main branch of your fork. Do you want to merge it first and then update this PR to base on that?

Or I can create a separate PR to linkedin:main

@ByronHsu thoughts?

**Comment by winglian** [comment]
@yubofredwang if your PR captures all the changes, I'm happy to have your PR supersede mine. thanks!

**Comment by ByronHsu** [comment]
@yubofredwang there are few conflicts
